### PR TITLE
Create Registration Form Preview Button

### PR DESCRIPTION
**Pull request by nkhur**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from Notion)
**Create Registration Form Preview Button**
The registration form editor on the admin side does not display the final version of the form. 

Add a button that will display the registration form exactly as shown to users. It will only render a frontend view of the form. It inputs should all be disabled and so should the submit button. As an extra safeguard, remove all backend logic from the submit button too.

This will allow admins to confirm their changes and ensure everything looks correct.

### Code Review Preparation
- [x] I have performed a self-review and reviewed the changed files
- [x] My code follows the project's style guidelines and passes linting
- [x] I have added comments in hard-to-understand areas and updated documentation

### Testing
- [x] I have added relevant tests and they pass locally
- [x] I have tested the changes manually, including edge cases

### Code Quality & Security
- [x] I have formatted my code and removed debugging/unused code
- [x] No TypeScript errors or linting issues
- [x] No sensitive information committed (API keys, passwords, etc.)
- [x] User inputs are validated and sanitized

## Screenshots/Screen recording
Editing View:
image

Preview:
image

## Summary by CodeRabbit

* **New Features**
  * Preview mode in the form builder with a Preview button to view sections as end-users will see them (compact/full variants, step navigation). Submission and validation are disabled in preview.

* **Style**
  * New preview UI styling and layout, updated button and icon styles (including a consistent font), refined delete/action button appearance, fixed heights and spacing, and improved disabled-state visuals for form controls.

### CODE REVIEW

**Comment by evtran4** [comment]
this is fantastic! great work navya i really like it

File: src/pages/FormBuilder/FormBuilder.module.css [hunk 1]
```diff
@@ -270,6 +272,73 @@
   gap: 0;
 }
 
+.contentHeaderRow {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 16px;
+}
+
```
[the remark this answers is not in the document]
Comment (evtran4) [reply src/pages/FormBuilder/FormBuilder.module.css]: This can be removed if not used anywhere

File: src/pages/FormBuilder/FormBuilder.module.css [hunk 2]
```diff
@@ -296,23 +365,43 @@
...
   border: 1px solid var(--gray-200);
   border-radius: var(--radius-sm);
   overflow: hidden;
 }
+
+.sectionTitleActionsDelete {
+  margin-left: 0;
+  padding-left: 0;
+}
+
```
[the remark this answers is not in the document]
Comment (evtran4) [reply src/pages/FormBuilder/FormBuilder.module.css]: same with this

**Review by evtran4 — CHANGES_REQUESTED** [review]
Please remove any unused code if necessary. Other than that it looks great!

**Review by evtran4 — APPROVED** [review]

**Comment by evtran4** [comment]
lgtm! great work
